### ISSUE TITLE
Task failing when trakt addon disabled for one user

### ISSUE THREAD

**Issue by toxic0berliner** [jellyfin/jellyfin-plugin-trakt#92]
Hello,
I have 2 users on my jellyfin instance, one (me) has a trakt account but the other don't. Since I added the second user the trakt task keeps failing.
I went and unchecked "synchronize collection" and "scrobble" for the other user, forced a reauth for me, tried several settings but to no avail : the task keeps failing...
Maybe I missed something or maybe the extension cannot cope with ignoring one user...
Thx in advance for any help.

**Comment by linetrimmer** [comment]
I am running into the same issue. I added two users to my server last night and now the Trakt plugin fails to sync (both the lmport playstates and Sync library scheduled tasks). The plugin worked fine a day ago—the only thing that has changed is the additional two users. 

I get the same error in my logs as the comment above.

**Comment by toxic0berliner** [comment]
I got tired of not getting my views synched to trackt so I deleted the other user and both trackt tasks are back to successful and it works fine, so it confirms we're just unable to disable the plugin for specific users only.

**Comment by linetrimmer** [comment]
Thanks for responding -- unfortunately deleting the two additional users didn't fix the error. I do have more users on my server that are not linked to a Trakt account (the plugin worked with these users on the server before adding the last two users), but I can't afford to delete these users from my server to test if it would start working then.

I've tried uninstalling/reinstalling the plugin, linking every user to a Trakt account, and wiping Trakt.xml and re-authorizing from scratch, but nothing works. The only way to get the task to complete without failing is to have 0 users linked to a Trakt account.

These are my error logs (I have 3 users linked to a Trakt account):


```
[2021-11-27 09:29:49.676 -05:00] [INF] [15] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks
[2021-11-27 09:29:50.904 -05:00] [INF] [22] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Sync library to trakt.tv"
[2021-11-27 09:29:50.948 -05:00] [ERR] [15] Emby.Server.Implementations.ScheduledTasks.TaskManager: Error
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncLibraryTask.SyncMovies(User user, TraktUser traktUser, ISplittableProgress`1 progress, CancellationToken cancellationToken)
   at Trakt.ScheduledTasks.SyncLibraryTask.SyncUserLibrary(User user, TraktUser traktUser, ISplittableProgress`1 progress, CancellationToken cancellationToken)
   at Trakt.ScheduledTasks.SyncLibraryTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
   at Emby.Server.Implementations.ScheduledTasks.ScheduledTaskWorker.ExecuteInternal(TaskOptions options)
[2021-11-27 09:29:50.949 -05:00] [INF] [15] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Sync library to trakt.tv" Failed after 0 minute(s) and 0 seconds
[2021-11-27 09:29:50.951 -05:00] [INF] [15] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks
[2021-11-27 09:29:53.585 -05:00] [INF] [29] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Import playstates from Trakt.tv"
[2021-11-27 09:29:53.627 -05:00] [ERR] [22] Trakt.ScheduledTasks.SyncFromTraktTask: Exception handled
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
[2021-11-27 09:29:53.629 -05:00] [ERR] [22] Trakt.ScheduledTasks.SyncFromTraktTask: Error syncing trakt data for user "******1"
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
[2021-11-27 09:29:53.673 -05:00] [ERR] [29] Trakt.ScheduledTasks.SyncFromTraktTask: Exception handled
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
[2021-11-27 09:29:53.674 -05:00] [ERR] [29] Trakt.ScheduledTasks.SyncFromTraktTask: Error syncing trakt data for user "******2"
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
[2021-11-27 09:29:53.755 -05:00] [ERR] [22] Trakt.ScheduledTasks.SyncFromTraktTask: Exception handled
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
[2021-11-27 09:29:53.756 -05:00] [ERR] [22] Trakt.ScheduledTasks.SyncFromTraktTask: Error syncing trakt data for user "******3"
System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
 ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore[TValue](JsonReaderState& readerState, Boolean isFinalBlock, ReadOnlySpan`1 buffer, JsonSerializerOptions options, ReadStack& state, JsonConverter converterBase)
   at System.Text.Json.JsonSerializer.ReadAsync[TValue](Stream utf8Json, Type returnType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at Trakt.Api.TraktApi.SendGetAllWatchedMoviesRequest(TraktUser traktUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.SyncTraktDataForUser(User user, Double currentProgress, CancellationToken cancellationToken, IProgress`1 progress, Double percentPerUser)
   at Trakt.ScheduledTasks.SyncFromTraktTask.Execute(CancellationToken cancellationToken, IProgress`1 progress)
[2021-11-27 09:29:53.756 -05:00] [INF] [22] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Import playstates from Trakt.tv" Completed after 0 minute(s) and 0 seconds
```

**Comment by haykkh** [comment]
Hello, have you tried [this](https://github.com/jellyfin/jellyfin-plugin-trakt/issues/79#issuecomment-881267918)? 
> I got it working now. Deleted the plugin first and the most important part: I deleted the Trakt.xml from Jellyfin/config/plugins/configurations. This file contained some elder information which caused the sync fail. After all I re-installed the trakt plugin, configured it and it is working now.


Seems to be a temporary fix until something more permanent is figured out.

**Comment by Shadowghost** [comment]
Should be fixed by #115

Some of the errors resulted from the user not bein authed anymore. We can#t do much in that case, the suer needs to be reauthenticated via the UI.
